bin/cheribsdtest/cheribsdtest_vm.c

```diff
@@ -2309,6 +2309,158 @@ CHERIBSDTEST(revoke_merge_quarantined,
 	cheribsdtest_success();
 }
 #undef NRES
+
+/*
+ * A simple test to confirm that revocation of a capability in a COW mapping
+ * affects only the caller's mapping.
+ */
+CHERIBSDTEST(cheri_revoke_cow_mapping,
+    "verify that revocation of a COW page triggers a copy",
+    .ct_check_skip = skip_need_cheri_revoke)
+{
+	void **block, **cap1, **cap2;
+	void *shadow, *torev;
+	ssize_t n;
+	size_t blocksz;
+	pid_t child;
+	int pd[2], res;
+	char ch, st[2];
+
+	/*
+	 * Use three pages for our heap.  The last page will be revoved.  The
+	 * first two pages contain a pointer into the third page; the first
+	 * page will be unmapped before revocation, while the second will remain
+	 * mapped.  This difference exercises different code paths in the
+	 * revoker.
+	 */
+	blocksz = 3 * PAGE_SIZE;
+	block = mmap(NULL, blocksz, PROT_READ | PROT_WRITE, MAP_ANON, -1, 0);
+	CHERIBSDTEST_VERIFY(block != MAP_FAILED);
+
+	torev = cheri_setbounds(block + 2 * PAGE_SIZE / sizeof(void *),
+	    PAGE_SIZE);
+	cap1 = cheri_setbounds(&block[0], PAGE_SIZE);
+	cap2 = cheri_setbounds(&block[PAGE_SIZE / sizeof(void *)], PAGE_SIZE);
+	*cap1 = *cap2 = cheri_andperm(torev, ~CHERI_PERM_SW_VMEM);
+
+	child = fork();
+	if (child == -1)
+		cheribsdtest_failure_errx("Fork failed; errno=%d", errno);
+	if (child == 0) {
+		/*
+		 * Quarantine the third page.
+		 */
+		CHERIBSDTEST_CHECK_SYSCALL(cheri_revoke_get_shadow(
+		    CHERI_REVOKE_SHADOW_NOVMEM, torev, &shadow));
+		memset(shadow, 0xff, cheri_getlen(shadow));
+
+		/*
+		 * Make sure that the first page is unmapped in the current
+		 * pmap.
+		 */
+		CHERIBSDTEST_CHECK_SYSCALL(msync(cap1, PAGE_SIZE,
+		    MS_INVALIDATE));
```

Comment:
Is it worth adding a note to the comment above for why you are using msync(MS_INVALIDATE) rather than munmap to unmap the page?